dattri/algorithm/influence_function.py

```diff
@@ -815,3 +815,194 @@ def _get_layer_wise_reps(self,
+            ValueError: If there are multiple checkpoints in `task`.
+        """
+        super().__init__(task, None, device)
+        if len(self.task.checkpoints) > 1:
+            error_msg = ("Received more than one checkpoint. "
+                         "Ensemble of EK-FAC is not supported.")
+            raise ValueError(error_msg)
+
+        if module_name is None:
+            # Select all linear layers by default
+            module_name = [
+                name for name, mod in self.task.model.named_modules()
+                if isinstance(mod, torch.nn.Linear)
+            ]
+        if not isinstance(module_name, list):
+            module_name = [module_name]
+
+        self.module_name = module_name
+
+        # Update layer_name corresponding to selected modules
+        self.layer_name = [name + ".weight" for name in self.module_name]
+
+        self.damping = damping
+        self.name_to_module = {
+            name: self.task.model.get_submodule(name) for name in module_name
+        }
+        self.module_to_name = {v: k for k, v in self.name_to_module.items()}
+
+        self.layer_cache = {}  # cache for each layer
+
+    def cache(
+        self,
+        full_train_dataloader: DataLoader,
+        max_iter: Optional[int] = None,
+    ) -> None:
+        """Cache the dataset and statistics for inverse FIM calculation.
+
+        Cache the full training dataset as other attributors.
+        Estimate and cache the covariance matrices, eigenvector matrices
+        and corrected eigenvalues based on the samples of training data.
+
+        Args:
+            full_train_dataloader (DataLoader): The dataloader
+                with full training samples for inverse FIM calculation.
+            max_iter (int, optional): An integer indicating the maximum number of
+                batches that will be used for estimating the the covariance matrices
+                and lambdas. Default to length of `full_train_dataloader`.
+        """
+        from dattri.func.fisher import (
+            estimate_covariance,
+            estimate_eigenvector,
+            estimate_lambda,
+        )
+
+        self._set_full_train_data(full_train_dataloader)
+
+        if max_iter is None:
+            max_iter = len(full_train_dataloader)
+
+        def _ekfac_hook(module: torch.nn.Module,
+                        inputs: Union[Tensor, Tuple[Tensor]],
+                        outputs: Union[Tensor, Tuple[Tensor]],
+            ) -> None:
+            """Hook function for caching the inputs and outputs of a module.
+
+            Args:
+                module (torch.nn.Module): The module to which the hook is registered.
+                inputs (Union[Tensor, Tuple[Tensor]]): The module input tensor(s).
+                outputs (Union[Tensor, Tuple[Tensor]]): The module output tensor(s).
+            """
+            # Unpack tuple outputs if necessary
+            if isinstance(inputs, tuple):
+                inputs = inputs[0]
```

Comment:
[bias issue] Here, we are using this `inputs` as the `a_prev` in our calculation for cov and lambda. While here we should append a `torch.ones` to the input to handle the bias.